awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -551,12 +635,12 @@ def create_if_needed(self, existing_item, new_item, endpoint, on_create=None, it
                 self.modify_associations(sub_endpoint, associations[association_type])
 
         # If we have an on_create method and we actually changed something we can call on_create
-        if on_create is not None and self.json_output['changed']:
+        if on_create is not None and self.json_output['created']:
```

Comment:
If you modify a project's URL or something like that, you still need to wait for update because it will auto-update in those cases.